src/Infrastructure/Field/tests/ESMF_FieldIOUTest.F90

```diff
@@ -213,16 +220,44 @@ program ESMF_FieldIOUTest
        iofmt=ESMF_IOFMT_NETCDF_64BIT_OFFSET,  &
        overwrite=.true.,  &
        status=ESMF_FILESTATUS_UNKNOWN, rc=rc)
-  write(failMsg, *) "Did not return ESMF_SUCCESS"
+  write(failMsg, *) "Did not return ESMF_SUCCESS: ", rc
   write(name, *) "Write Fortran array in Field"
 #if (defined ESMF_PIO && ( defined ESMF_NETCDF || defined ESMF_PNETCDF))
   call ESMF_Test((rc==ESMF_SUCCESS), name, failMsg, result, ESMF_SRCLINE)
 #else
   write(failMsg, *) "Did not return ESMF_RC_LIB_NOT_PRESENT"
   call ESMF_Test((rc==ESMF_RC_LIB_NOT_PRESENT), name, failMsg, result, ESMF_SRCLINE)
 #endif
+!  call ESMF_Finalize(endflag=ESMF_END_ABORT)
+!  stop
+
+  write(failMsg, *) ""
+  write(name, *) "Read data just written to field_r"
+  call ESMF_FieldRead(field_r, fileName="field.nc", rc=rc)
+#if (defined ESMF_PIO && ( defined ESMF_NETCDF || defined ESMF_PNETCDF))
+  call ESMF_Test((rc==ESMF_SUCCESS), name, failMsg, result, ESMF_SRCLINE)
+#else
+  write(failMsg, *) "Did not return ESMF_RC_LIB_NOT_PRESENT"
+  call ESMF_Test((rc==ESMF_RC_LIB_NOT_PRESENT), name, failMsg, result, ESMF_SRCLINE)
+#endif
+
+
+  call ESMF_FieldGet(field_r, localDe=0, farrayPtr=Farray_r, &
+      exclusiveLBound=exclusiveLBound, &
+      exclusiveUBound=exclusiveUBound, rc=rc)
+  write(failMsg, *) "Did not return ESMF_SUCCESS"
+  write(name, *) "Get Farray_r from field"
+  call ESMF_Test((rc.eq.ESMF_SUCCESS), name, failMsg, result, ESMF_SRCLINE)
+  diff = 0.0
+  do j=exclusiveLBound(2),exclusiveUBound(2)
+  do i=exclusiveLBound(1),exclusiveUBound(1)
+     diff = diff + abs(Farray_w(i,j) - Farray_r(i,j))
+  enddo
+  enddo
+  write(failMsg, *) "Field read did not match field write"
+  write(name, *) "Compare field read to field write"
+  call ESMF_Test(diff == 0.0, name, failMsg, result, ESMF_SRCLINE)
```

Comment:
Looks like this is a new test.  Can we guarantee 0.0 diff in general, or do we need room for roundoff-level diffs?

Reply:
It should be exact.